meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -115,6 +117,61 @@ public void create(String key, Object data, MetaClientInterface.EntryMode mode)
     }
   }
 
+  @Override
+  public void createFullPath(String key, Object data, MetaClientInterface.EntryMode mode) {
+
+    boolean retry;
+    MetaClientInterface.EntryMode parentMode = (EntryMode.EPHEMERAL.equals(mode) ?
+        EntryMode.PERSISTENT : mode);
+    do {
+      retry = false;
+      try {
+        create(key, data, mode);
+      } catch (MetaClientNoNodeException e) {
+        retry = true;
+        String parentPath = getZkParentPath(key);
+        try {
+          createFullPath(parentPath, null, parentMode);
+        } catch (MetaClientNodeExistsException e1) {
```

Comment:
Probably we dont need catch here?

Reply:
Good point, it would just be excessive logging. Final error thrown/caught should be enough